dotrix_core/src/renderer.rs

```diff
@@ -100,7 +101,25 @@ impl Renderer {
         height: u32,
         layers: &'a [&'a [u8]],
     ) {
-        buffer.load(self.backend(), width, height, layers);
+        self.load_texture_buffer_with_usage(
+            buffer,
+            width,
+            height,
+            layers,
+            wgpu::TextureUsages::TEXTURE_BINDING | wgpu::TextureUsages::COPY_DST,
+        );
+    }
+
+    /// Loads the texture buffer to GPU with usages
+    pub fn load_texture_buffer_with_usage<'a>(
```

Comment:
Yes I was thinking about something similar where we use the tie things more into the underlying asset id more than so that reloads can be handelled more gracefully.

For example at the moment when I resize my screen I have to resize the depth and light map textures, this means I have to resize the texture, call pipeline.buffer.reload and drop the shader so that the buffers used the new texture. It would be nice if the pipeline.buffers used Id<Texture> and knew when they were invalid and needed reloading.

Reply:
yeah, that is a good point too